src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -2218,6 +2218,31 @@ public static bool IsIsOrAsContext(this SyntaxTree syntaxTree, int position, Syn
             // is/as are valid after expressions.
             if (token.IsLastTokenOfNode<ExpressionSyntax>())
             {
+                var expression = token.GetAncestor<ExpressionSyntax>();
+                var symbols = semanticModel.GetSymbolInfo(expression).GetBestOrAllSymbols();
```

Comment:
❓ Is this pattern (ancestor expression &rarr; `GetBestOrAllSymbols`) used elsewhere in keyword recommenders? If so, is the result cached? (Trying to determine if this may cause performance concerns.)